source/CollisionSet.cpp

```diff
@@ -164,25 +167,51 @@ void CollisionSet::Finish()
 
 
 
-// Get the first object that collides with the given projectile. If a
-// "closest hit" value is given, update that value.
-Body *CollisionSet::Line(const Projectile &projectile, double *closestHit) const
+// Get the first (closest) collision for the given projectile.
+Collision CollisionSet::Line(const Projectile &projectile) const
 {
 	// What objects the projectile hits depends on its government.
 	const Government *pGov = projectile.GetGovernment();
 
 	// Convert the start and end coordinates to integers.
 	Point from = projectile.Position();
 	Point to = from + projectile.Velocity();
-	return Line(from, to, closestHit, pGov, projectile.Target());
+	return Line(from, to, pGov, projectile.Target());
 }
 
 
 
-// Check for collisions with a line, which may be a projectile's current
+// Get all collisions for the given projectile. Collisions are not necessarily
+// sorted by distance.
+const vector<Collision> &CollisionSet::LineAll(const Projectile &projectile) const
+{
+	// What objects the projectile hits depends on its government.
+	const Government *pGov = projectile.GetGovernment();
+
+	// Convert the start and end coordinates to integers.
```

Comment:
This comment seems wrong? Point stores doubles iirc

Reply:
Yeah, not sure what this comment meant. It's copied from line 176 (and that function could probably just call this one).

Reply:
Updated the comment.